openstf_dbc/services/predictor.py

```diff
@@ -2,36 +2,137 @@
 #
 # SPDX-License-Identifier: MPL-2.0
 
+from enum import Enum
+from typing import List, Optional, Tuple, Union
+
+import pandas as pd
 from openstf_dbc.data_interface import _DataInterface
+from openstf_dbc.services.weather import Weather
+from openstf_dbc.utils.utils import get_datetime_index
+
+
+class PredictorGroups(Enum):
+    MARKET_DATA = "market_data"
+    WEATHER_DATA = "weather_data"
+    LOAD_PROFILES = "LOAD_PROFILES"
 
 
 class Predictor:
-    def get_apx(self, datetime_start, datetime_end):
+    def get_predictors(
+        self,
+        datetime_start,
+        datetime_end,
+        forecast_resolution: Optional[str] = None,
+        location: Union[str, Tuple[float, float]] = None,
+        predictor_groups: Optional[List[str]] = None,
+    ):
+        """Get predictors.
+
+        Get predictors for a given datetime range. Optionally predictor groups can be
+        selected. If the WEATHER_DATA group is included a location is required.
+
+        Args:
+            location (Union[str, Tuple[float, float]], optional): Location (for weather data).
+                Defaults to None.
+            predictor_groups (Optional[List[str]], optional): The groups of predictors
+                to include. Defaults to None.
```

Comment:
'defaults to None, which results in retrieving data for all PredictorGroups'

Reply:
I have updated the docstring!